doc/mssqlserver/doc.md

```diff
@@ -0,0 +1,145 @@
+  * Default Value: `NONE`
+  * Required: True if `MSSQLSERVER_URL` is not set.
+  * System Property: `org.jboss.eap.datasources.mssqlserver.database`
+
+* `MSSQLSERVER_HOST` (or `MSSQLSERVER_SERVICE_HOST`)
+
+  * Description: Defines the database server’s host name or IP address to be used in the datasource’s `connection-url` property.
+  * Default Value: `NONE`
+  * Required: True if `MSSQLSERVER_URL` is not set.
+  * System Property: `org.jboss.eap.datasources.mssqlserver.host`
+
+* `MSSQLSERVER_PORT` (or `MSSQLSERVER_SERVICE_PORT`)
+
+  * Description: Defines the database server’s port to be used in the datasource’s `connection-url` property. 
+  * Default Value: `NONE`
+  * Required: True if `MSSQLSERVER_URL` is not set.
+  * System Property: `org.jboss.eap.datasources.mssqlserver.port`
+
+* `MSSQLSERVER_PASSWORD`
+
+  * Description: Defines the password for the datasource.
+  * Default Value: `NONE`
+  * Required: True
+  * System Property: `org.jboss.eap.datasources.mssqlserver.password`
+
+* `MSSQLSERVER_URL`
+
+  * Description: Defines the connection URL for the datasource. 
+  * Default Value: `jdbc:sqlserver://${MSSQLSERVER_HOST}:${MSSQLSERVER_PORT};DatabaseName=${MSSQLSERVER_DATABASE}`
+  * Required: True if `MSSQLSERVER_PORT/HOST/DATABASE` are not set.
+  * System Property: `org.jboss.eap.datasources.mssqlserver.connection-url`
+
+* `MSSQLSERVER_USER`
+
+  * Description: Defines the username for the datasource. 
+  * Default Value: `NONE`
+  * Required: True
+  * System Property: `org.jboss.eap.datasources.mssqlserver.user-name`
+
+Optional configuration
+==============
+
+* `MSSQLSERVER_BACKGROUND_VALIDATION`
+
+  * Description: When set to true database connections are validated periodically in a background thread prior to use. Defaults to false, meaning the `validate-on-match` method is enabled by default instead.  
+  * Default Value: `false`
+  * System Property: `org.jboss.eap.datasources.mssqlserver.background-validation`
+
+
+* `MSSQLSERVER_BACKGROUND_VALIDATION_MILLIS`
+
+  * Description: Specifies frequency of the validation, in milliseconds, when the `background-validation` database connection validation mechanism is enabled.    
+  * Default Value: `10000`
+  * System Property: `org.jboss.eap.datasources.mssqlserver.background-validation-millis`
+
+* `MSSQLSERVER_CONNECTION_CHECKER`
+
+  * Description: Specifies a connection checker class that is used to validate connections. Valid value: `org.jboss.jca.adapters.jdbc.extensions.mssql.MSSQLValidConnectionChecker`
+  * Default Value: `org.jboss.jca.adapters.jdbc.extensions.novendor.NullValidConnectionChecker`
+  * System Property: `org.jboss.eap.datasources.mssqlserver.valid-connection-checker-class-name`
+
+* `MSSQLSERVER_DATASOURCE`
+
+  * Description: Datasource name used in the `jndi-name`.
+  * Default Value: `MSSQLServerDS`
+  * System Property: `org.jboss.eap.datasources.mssqlserver.datasource`
+
+
+* `MSSQLSERVER_ENABLED`
```

Comment:
why do we need that property?
If we don't want the datasource, we just have to not provision its layer...

Reply:
This allow allow to a build a bootable JAR but disable the DS for some context/reason. I feel that it can give flexibility.